scripts/test/maven/runTests.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+# Copyright 2022-2024 Crown Copyright
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+unset CDPATH
+
+THIS_DIR=$(cd "$(dirname "$0")" && pwd)
+SCRIPTS_DIR=$(cd "$THIS_DIR" && cd ../.. && pwd)
+MAVEN_DIR=$(cd "$SCRIPTS_DIR" && cd ../java && pwd)
+
+if [ "$#" -ne 3 ]; then
+  echo "Usage: $0 <vpc> <subnet> <results-bucket>"
+  exit 1
+fi
+
+source "$SCRIPTS_DIR/functions/timeUtils.sh"
+source "$SCRIPTS_DIR/functions/systemTestUtils.sh"
+
+VPC=$1
+SUBNETS=$2
+RESULTS_BUCKET=$3
+START_TIMESTAMP=$(record_time)
+TEST_NAME=QuickSystemTests
+OUTPUT_DIR="/tmp/sleeper/${TEST_SUITE_NAME}Tests/$START_TIME"
+SHORT_ID="i-$(uuidgen -t | cut -d'-' -f 1)"
+
+mkdir -p "$OUTPUT_DIR"
+VERSION=$(cat "$SCRIPTS_DIR/templates/version.txt")
+SYSTEM_TEST_JAR="$SCRIPTS_DIR/jars/system-test-${VERSION}-utility.jar"
+
+END_EXIT_CODE=0
+
+runMavenQuickSystemTests() {
+    echo "-------------------------------------------------------------------------------"
+    echo "Running quick system tests"
+    echo "-------------------------------------------------------------------------------"
+
+    "$SCRIPTS_DIR/test/maven/buildDeployTest.sh" "$SHORT_ID" "$VPC" "$SUBNETS" > "$OUTPUT_DIR/$TEST_NAME.log"
+    echo -e "Test run complet \nSee $OUTPUT_DIR/$TEST_NAME.log for results"
+
+    EXIT_CODE=$?
```

Comment:
I think this assignment needs to be before the echo statement above, else it will take the exit code from the echo command and not the `buildDeployTest` call

Reply:
Moved the assignment